app/src/main/java/org/oppia/app/profile/ProfileChooserFragmentPresenter.kt

```diff
@@ -126,16 +145,65 @@ class ProfileChooserFragmentPresenter @Inject constructor(
   }
 
   private fun bindAddView(binding: ProfileChooserAddViewBinding, @Suppress("UNUSED_PARAMETER") model: ProfileChooserUiModel) {
+
+    chooserViewModel.addViewVisibility(binding, whichViewAdd)
+
     binding.root.setOnClickListener {
       if (chooserViewModel.adminPin.isEmpty()) {
         activity.startActivity(
-          AdminPinActivity.createAdminPinActivityIntent(activity, chooserViewModel.adminProfileId.internalId, selectUniqueRandomColor())
+          AdminPinActivity.createAdminPinActivityIntent(
+            activity,
+            chooserViewModel.adminProfileId.internalId,
+            selectUniqueRandomColor()
+          )
         )
       } else {
         activity.startActivity(
-          AdminAuthActivity.createAdminAuthActivityIntent(activity, chooserViewModel.adminPin, selectUniqueRandomColor())
+          AdminAuthActivity.createAdminAuthActivityIntent(
+            activity,
+            chooserViewModel.adminPin,
+            selectUniqueRandomColor()
+          )
         )
       }
     }
+
+  }
+
+  private fun getProfileEverCreated(): LiveData<Boolean> {
+    return Transformations.map(
+      profileManagementController.getWasProfileEverAdded(),
+      ::processProfileEverCreated
+    )
   }
+
+  private fun subscribetoProfileEverCreated(
+    binding: ProfileChooserFragmentBinding
+  ) {
+    wasProfileEverCreatedLiveData.observe(fragment, Observer<Boolean> {
+      when (it) {
+        true -> {
+          binding.profileRecyclerView.layoutManager =
+            GridLayoutManager(activity.applicationContext, 2)
+          whichViewAdd = true
+        }
+
+        false -> {
+          binding.profileRecyclerView.layoutManager =
+            LinearLayoutManager(activity.applicationContext)
+          whichViewAdd = false
+        }
+
+      }
+    })
+  }
+
+  private fun processProfileEverCreated(boolean: AsyncResult<Boolean>): Boolean {
+    if (boolean.isFailure()) {
+      Log.e("TopicFragment", "Failed to retrieve topic", boolean.getErrorOrNull()!!)
```

Comment:
Use `logger` instead of `Log` you can check the usage of `logger` in other presenters. Also, the tag should be `ProfileChooserFragment` instead of `TopicFragment` and even the `Failed to retrieve topic` is incorrect.

Reply:
Okay, done :)